### PR TITLE
Reduce Travis build time

### LINKED ISSUE #9430: Return composer.lock to the repo
- Much faster composer installs (assets plugin is particularly slow)
- Faster Travis builds and zip releases
- Stop causing PR tests to fail due to changes in unrelated dependencies
- Stop causing PR tests to fail due to GitHub API limits (we're working around this now but still occasionally hit it)
- Devs get predictable installs
- Easier to support zip releases because we can actually duplicate code
- (anecdotal) Drupal commits its lock.

Else I think we should require specific versions of assets at least.

### PR DESCRIPTION
**Pull request by hypeJunction**
Success of a build depends on the state of vendor libraries. We need make
sure that vendor libraries in the successful build correspond to the
vendor libraries distributed to end users.

This also adds the benefit of faster Travis builds, Scrutinizer support
(which currently fails due to the composer asset plugin dependency).

Fixes #9430

### CODE REVIEW

**Comment by hypeJunction** [comment]
Build time went from 12min to 9min to 8min

**Comment by hypeJunction** [comment]
Build time is down to 3min. Scrutinizer was able to generate it's own code coverage: https://scrutinizer-ci.com/g/Elgg/Elgg/inspections/0ca154a4-03cb-421f-9b46-bcc858999190

**Comment by hypeJunction** [comment]
Adding git depth saves about 20 seconds on cloning the repo

**Comment by hypeJunction** [comment]
So, we got a 400% increase in build speed. I think that's as good as it gets.

File: .travis.yml [hunk 1]
```diff
@@ -95,3 +104,9 @@ notifications:
 env:
   global:
      secure: "fdpCjdC0Qp/ZJqtrCHE4I/tHXWF1sORftm6khd6geqK7d4qWzIh6HzNN2BlF+2m4nyuyxo2wzPm/oGoqogVpBgrzpQ7SZl7h2/wzgs2C/k39sFGyDisLesTM5DhBDJWcomyqtcnQmKn340Z9KOxiHAt4FOj2FZVN5+tIO5j3Cks="
+
+## Cache dependencies
```
Comment (mrclay) [inline .travis.yml]: What does this do?

**Comment by mrclay** [comment]
💯 🎉  Can't wait to say goodbye to vendor mismatches between my local and Travis.

**Comment by mrclay** [comment]
We have 3 or 4 more months supporting 1.12, why shouldn't we do this there?

File: .travis.yml (hunk 1 above)
Comment (hypeJunction) [inline .travis.yml]: Caches downloaded dependencies, so they are loaded from cache instead of full download

**Comment by hypeJunction** [comment]
I already said good bye to 1.12. But feel free to hijack

File: .scrutinizer.yml [hunk 2]
```diff
@@ -1,27 +1,66 @@
+build:
+    dependencies:
+        before:
+            - 'composer global require "fxp/composer-asset-plugin:~1.1.1"'
```
Comment (hypeJunction) [inline .scrutinizer.yml]: I don't think it matters, it will grab latest 1.1.x.

**Comment by hypeJunction** [comment]
I just looked into cherry picking this into 1.12, and conflicts are nightmare.

**Comment by mrclay** [comment]
Rebase and go for it.

**Pull request merged**